debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/ObjectMonitor_V2.java

```diff
@@ -0,0 +1,322 @@
+		if(isInflated) {
+			if(OmrBuildFlags.OMR_THR_THREE_TIER_LOCKING) {
+				J9ThreadPointer thread = J9ThreadMonitorHelper.getBlockingField(monitor);
+				while(thread.notNull()) {
+					J9VMThreadPointer vmThread = J9ThreadHelper.getVMThread(thread);
+					if(vmThread.notNull()) {
+						blockedThreads.add(vmThread);
+					}
+					thread = thread.next();
+				}
+			} else {
+				List<J9VMThreadPointer> list = getBlockedThreads(object);
+				if(list != null) {
+					blockedThreads.addAll(list);
+				}
+			}
+		} else {
+			// For consistency we walk the thread list and check for blocking objects
+			// rather than do lockword.allBitsIn(OBJECT_HEADER_LOCK_FLC); as the lockword
+			// is set slightly later (the thread may spin first). See: CMVC 201473
+			List<J9VMThreadPointer> list = getBlockedThreads(object);
+			if(list != null) {
+				blockedThreads.addAll(list);
+			}
+		}
+	}
+
+	private void initializeLockword() throws CorruptDataException
+	{
+		lockword = J9ObjectMonitorPointer.NULL;
+
+		// TODO : why isn't there a cast UDATA->IDATA?
+		IDATA lockOffset = new IDATA(J9ObjectHelper.clazz(object).lockOffset());
+		// TODO : why isn't there an int/long comparison
+		if(lockOffset.gte(new IDATA(0))) {
+			lockword = ObjectMonitorReferencePointer.cast(object.addOffset(lockOffset.longValue())).at(0);
+		} else {
+			if (j9objectMonitor.notNull()) {
+				lockword = j9objectMonitor.alternateLockword();
+			}
+		}
+
+		if(lockword.notNull()) {
+			isInflated = lockword.allBitsIn(OBJECT_HEADER_LOCK_INFLATED);
+			if (isInflated) {
+				/* knowing it's inflated we could extract the J9ObjectMonitor doing:
+				 *
+				 * 		J9ObjectMonitorPointer.cast(lockword.untag(OBJECT_HEADER_LOCK_INFLATED));
+				 *
+				 *...but we already have it so why bother
+				 */
+				monitor = J9ThreadAbstractMonitorPointer.cast(j9objectMonitor.monitor());
+			}
+		}
+	}
+
+	private static List<J9VMThreadPointer> getBlockedThreads(J9ObjectPointer blockingObject) throws CorruptDataException
+	{
+		// Repeatedly walking the thread list could get expensive.
+		// Do a single walk and cache all the results.
+		if(blockedThreadsCache == null) {
+			blockedThreadsCache = new HashMap<J9ObjectPointer, List<J9VMThreadPointer>>();
+			GCVMThreadListIterator iterator = GCVMThreadListIterator.from();
+			while (iterator.hasNext()) {
+				J9VMThreadPointer vmThread = iterator.next();
+				if(vmThread.publicFlags().allBitsIn(J9Consts.J9_PUBLIC_FLAGS_THREAD_BLOCKED)) {
+					J9ObjectPointer object = vmThread.blockingEnterObject();
+					if(object.notNull()) {
+						List<J9VMThreadPointer> list = blockedThreadsCache.get(object);
+						if(list == null) {
+							list = new ArrayList<J9VMThreadPointer>();
+							blockedThreadsCache.put(object, list);
+						}
+						list.add(vmThread);
+					}
+				}
+			}
+		}
+		return blockedThreadsCache.get(blockingObject);
+	}
+
+	@Override
+	public int hashCode() {
+		return super.hashCode();
+	}
```

Comment:
Why not simply inherit this behavior?

Reply:
That behavior was copied directly from `ObjectMonitor_V1.java` which does the exact same thing so I don't know the reasoning behind it. Should I get rid of it?

Reply:
I don't think it helps, so I think it should be removed.

Reply:
Okay, I'll remove it.